justfile

```diff
@@ -85,8 +85,12 @@ install-toolchain:
 	rustup component add llvm-tools-preview --toolchain "$(cat rust-toolchain)"
 	cargo install cargo-binutils
 
+# The following line gives highlighting on vim
+# vim: set ft=make :
+
+# Run a firwmare with benchmark settings. The firmware should call for benchmark printing before exiting
 benchmark firmware=benchmark iterations=default_iterations:
-	cargo run --package runner -- run -v --firmware {{firmware}} --benchmark --benchmark-iterations {{iterations}}
+	cargo run --package runner -- run -v --config {{qemu_virt_benchmark}} --firmware {{firmware}} --benchmark --benchmark-iterations {{iterations}}
```

Comment:
Is `--benchmark-iterations` overwriting `nb_iter` in the config?

I am just thinking that we should probably keep the same names for consistency, but we can do that in a separate commit/PR.

Reply:
No, it's two different things:

- `--benchmark-iterations` is the times the runner runs the entire triplet (miralis, payload and firmware) to compute average on the entire thing at the ends. It is only usefull for qemu benchmarking.
- `nb_iter` is an argument that is passed to our custom firmwares (that benchmarks only one thing many times). It can be used differently between firmwares (e.g. run `nb_iter` instructions, or `nb_iters` times a complex function or ecall). It is also used for building these firmwares in order to run on the board.

We can remove the first one and the just benchmark command if we don't want to benchmark (or not on many iterations) on qemu anymore.

Reply:
I see, that make sense thanks :)

Yes I am wondering if we want to keep the benchmarking capabilities for QEMU, we saw from experience with the board that the QEMU numbers don't make sense. Running the benchmarks on QEMU is still useful to test the code path and ensure nothing is broken, but it's just that the benchmark results don't make any sense so I am not sure if it makes sense to keep `--benchmark-iterations` for instance considering that we can't do anything with numbers from QEMU (contrary to what I hopped initially...). What do you think? Anyway, if we change something it can be done in a different PR.

Reply:
If the number doesn't make any sens, it is better to remove ´--benchmark-iterations´, and the ´just benchmark´ command, and only keep the ´--benchmark´ argument for the runner in the tests to test benchmark code. It will be simpler and more understandable then.